src/runtime/node/wasi/wasi.ts

```diff
@@ -0,0 +1,12 @@
+import type wasi from "node:wasi";
+
+export class WASI implements wasi.WASI {
```

Comment:
similar to the `vm` module, I think the constructor of WASI should throw as not implemented. I can't think of why noop implementation would help some applications to run - instead we'll see people confused when their apps fail at runtime with very weird downstream errors.